src/server/browserContext.ts

```diff
@@ -179,6 +183,63 @@ export abstract class BrowserContext extends SdkObject {
     await this._doExposeBinding(binding);
   }
 
+  async fetch(params: types.FetchOptions): Promise<{response?: network.FetchResponse, error?: string}> {
+    try {
+      const cookies = await this.cookies(params.url);
+      const valueArray = cookies.map(c => `${c.name}=${c.value}`);
+      const clientCookie = params.headers?.['cookie'];
+      if (clientCookie)
+        valueArray.unshift(clientCookie);
+      const cookieHeader = valueArray.join('; ');
+      if (cookieHeader) {
+        if (!params.headers)
+          params.headers = {};
+        params.headers['cookie'] = cookieHeader;
+      }
+      if (!params.method)
+        params.method = 'GET';
+      let agent;
+      if (this._options.proxy) {
+        const proxyOpts = url.parse(this._options.proxy.server);
+        if (this._options.proxy.username)
+          proxyOpts.auth = `${this._options.proxy.username}:${this._options.proxy.password || ''}`;
+        agent = new HttpsProxyAgent(proxyOpts);
+      }
+
+      const request = new network.FetchRequest(this, params.url, params.method, params.headers, params.postData);
+      // TODO: set user agent
+      const response = await fetch(params.url, {
+        method: params.method,
+        headers: params.headers,
+        body: params.postData,
+        agent});
+      const setCookies = response.headers.raw()['set-cookie'];
+      if (setCookies) {
+        const cookies: types.SetNetworkCookieParam[] = [];
+        for (const header of setCookies) {
+          // Decode cookie value?
+          const cookie: types.SetNetworkCookieParam | null = parseCookie(header);
+          if (!cookie)
+            continue;
+          if (!cookie.domain)
+            cookie.url = params.url;
+          cookies.push(cookie);
+        }
+        if (cookies.length)
+          await this.addCookies(cookies);
```

Comment:
Just as an idea, I think users might want to see this optional. Same goes for the cokie: headers.

Reply:
I'll add these options in a follow-up PR once we settle on the base API.